src/XMonad/Core.hs

```diff
@@ -365,12 +366,33 @@ instance Show (Layout a) where show (Layout l) = show l
 --
 -- User-extensible messages must be a member of this class.
 --
-class Typeable a => Message a
+class Typeable a => Message a where
+    someMessage :: a -> SomeMessage -- NOT EXPORTED; use `SomeMessage`
+    someMessage = AMessage
 
 -- |
 -- A wrapped value of some type in the 'Message' class.
 --
-data SomeMessage = forall a. Message a => SomeMessage a
+data SomeMessage = forall a. Message a => AMessage a
+    deriving (Typeable)
```

Comment:
GHC derives `Typeable` automatically for every type now, so this is probably superfluous

Reply:
That's good news.

...and `derive (Typeable)` is now gone.